data_structures/src/builders.rs

```diff
@@ -66,7 +66,7 @@ impl Message {
                 version: PROTOCOL_VERSION,
                 timestamp: get_timestamp(),
                 capabilities: CAPABILITIES,
-                sender_address: to_address(sender_addr),
+                sender_address: sender_addr.map(to_address),
```

Comment:
Maybe we can serialize `None` as `SocketAddr::default()` to avoid that ugly `impl ProtobufConvert`

```suggestion
                sender_address: sender_addr.unwrap_or_default(),
```